lib/src/webview.dart

```diff
@@ -566,3 +593,16 @@ class _WebviewState extends State<Webview> {
     _cursorSubscription?.cancel();
   }
 }
+
+extension WebviewHostResourceAccessKindExt on WebviewHostResourceAccessKind {
```

Comment:
For consistency reasons, we just pass the corresponding enum member's index via the method channel and have the same enum on the C++ side which is kept in sync with the Dart enum then. (see https://github.com/jnschulze/flutter-webview-windows/blob/91127a465bffeccf4d72d27f3c9ab8690a374449/windows/webview.h#L13-L29)

Reply:
Good Point! I seem to completely overlooked that while glancing over the code.

I'll added it (and rearranged the order to match [`COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND`](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2#corewebview2_host_resource_access_kind).

Would be assigning the values from `COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND` to the new enum be a good option?

Reply:
I would create an enum whose values are independent of the WebView2 enum values. I'd then cast the int received in the method channel handler in `webview_bridge.cc` to the internal enum `WebviewHostResourceAccessKind` (which needs to be kept in sync with your Dart enum). The `switch` mapping the internal enum to the WebView2 enum should be moved to the `Webview::SetVirtualHostNameMapping` implementation.